test/irmin-pack/test_gc.ml

```diff
@@ -0,0 +1,467 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+open Common
+
+let src = Logs.Src.create "tests.gc" ~doc:"Test gc"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let test_dir = "_build"
+
+let fresh_name =
+  let c = ref 0 in
+  fun () ->
+    incr c;
+    let name = Filename.concat test_dir ("test-gc" ^ string_of_int !c) in
+    name
```

Comment:
I sort-of worry that these names aren't actually fresh... what happens if we run the tests (so, lots of _build/test-gcNNNN dirs) then run the tests again? Do the dirs get reused? And if so, does the contents of the dirs mess up the tests? One option is just to use Sys.file_exists or similar, and keep incrementing the counter till we hit a fresh name. Or maybe just add a note that we expect to run the tests just once, on a _build dir with no test-gcNNNN subdirs?